turbomoleio/core/base.py

```diff
@@ -0,0 +1,252 @@
+                                              vesta=True)
+        elif periodicity == 3:
+            if len(cell) != 6:
+                raise ValueError('The $cell data group should contain 6 numbers for 3D periodic systems')
+            lattice = Lattice.from_parameters(a=bohr_to_ang*cell[0],
+                                              b=bohr_to_ang*cell[1],
+                                              c=bohr_to_ang*cell[2],
+                                              alpha=cell[3],
+                                              beta=cell[4],
+                                              gamma=cell[5],
+                                              vesta=True)
+        else:
+            raise RuntimeError('Periodicity should be one of 1, 2 or 3.')
+        mol = Structure(lattice=lattice, species=species, coords=coords, coords_are_cartesian=True,
+                        to_unit_cell=False)
+    elif lattice_string is not None:
+        lattice_string = lattice_string.strip()
+        periodicity = int(periodic_string.strip())
+        lattice_numbers = np.array([[float(ln) for ln in line.split()] for line in lattice_string.splitlines()])
+        if periodicity == 1:
+            if lattice_numbers.shape != (1, 1):
+                raise ValueError('The lattice_numbers should contain 1 number for 1D periodic systems')
+            yrange = np.max(coords[:, 1])-np.min(coords[:, 1])
+            zrange = np.max(coords[:, 2])-np.min(coords[:, 2])
+            lattice = Lattice.orthorhombic(a=bohr_to_ang*lattice_numbers[0][0],
+                                           b=bohr_to_ang*yrange+periodic_extension,
+                                           c=bohr_to_ang*zrange+periodic_extension)
+        elif periodicity == 2:
+            if lattice_numbers.shape != (2, 2):
+                raise ValueError('The lattice_numbers should contain 2x2 numbers for 2D periodic systems')
+            zrange = np.max(coords[:, 2])-np.min(coords[:, 2])
+            lat_matrix = [[bohr_to_ang*lattice_numbers[0, 0], bohr_to_ang*lattice_numbers[0, 1], 0.0],
+                          [bohr_to_ang*lattice_numbers[1, 0], bohr_to_ang*lattice_numbers[1, 1], 0.0],
+                          [0.0, 0.0, bohr_to_ang*zrange+periodic_extension]]
+            lattice = Lattice(lat_matrix)
+        elif periodicity == 3:
+            if lattice_numbers.shape != (3, 3):
+                raise ValueError('The lattice_numbers should contain 3x3 numbers for 3D periodic systems')
+            lattice = Lattice(bohr_to_ang*np.array(lattice_numbers))
+        else:
+            raise RuntimeError('Periodicity should be one of 1, 2 or 3.')
+        mol = Structure(lattice=lattice, species=species, coords=coords, coords_are_cartesian=True,
+                        to_unit_cell=False)
+    else:
+        mol = Molecule(species, coords)
+    CoordMolecule = namedtuple("CoordMolecule", ["molecule", "frozen_indices"])
```

Comment:
should we change the name of this nametuple? it should not break anything I suppose.

Reply:
Yes. I will do: CoordSystem and have "system" instead of molecule. If you think this is not good, we can fall back to something else after

Reply:
Well actually it is not a "System" it is either a Molecule object or a Structure ... I'll put molecule_or_structure instead.

Reply:
I think it is fine, but I would like to find a better name than `molecule_or_structure` for these kind of variables in the future. For reference, in pymatgen in the few places where the accepted type is `Union[Structure, Molecule]` the variable is named  `structure`